src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -721,6 +725,9 @@ public void recover(Engine.RecoveryHandler recoveryHandler) throws EngineExcepti
     }
 
     public void failShard(String reason, Throwable e) {
+        if (engineClosed()) {
```

Comment:
why do we need to check this here I think we should never check this and just call fail on it?